### ISSUE TITLE
Create MIME renderer for QZVs in JupyterLab

### ISSUE THREAD

**Issue by ElDeveloper** [qiime2/qiime2#806]
**Addition Description**

QIIME2 currently supports Jupyter integration through the Visualization object. However, JupyterLab does not support viewing a QZV file by simply clicking on it in the file browser. Fortunately, Jupyter allows custom handling of [MIME types](https://github.com/jupyterlab/extension-examples/tree/main/mimerenderer) via a TypeScript extension. It would be very useful to implement this feature especially when many artifacts are being created and analyzed.

**Current Behavior**
When clicking on a QZV file in the JupyterLab file browser, Jupyter displays a default error message, indicating that it cannot interpret the file because it is not UTF-8 encoded.

**Proposed Behavior**
Clicking on a QZV file in the JupyterLab file browser would open the file in a similar way to how `qiime tools view` works.

**Questions**
1. Can this even be included in the qiime2/qiime2 repository or will it need its own repository?

**Comments**
1. It seems that q2view already handles much of the process of opening, parsing, and serving QZV files via JavaScript. However, I am not familiar enough with the codebase to determine how compatible this would be with Jupyter.

**References**
1. https://github.com/jupyterlab/extension-examples/tree/main/mimerenderer

**Comment by gregcaporaso** [comment]
This would also be _really_ handy for the new Docker workshop containers. We had some content on this topic in the q2-ITN workshop [here](https://q2-cat-book.readthedocs.io/en/latest/08-workshop-infrastructure.html#notebook-qzv-viewing), but this would be a nicer way to handle it.

**Comment by ebolyen** [comment]
I think we'll need to still use a server-extension to host the individual pages under `/data/` as I don't think we can install a service worker to handle this entirely client side (I could be wrong, but it definitely would exceed a MIME renderer).

So I expect this looks like a server-extensions (https://github.com/jupyterlab/extension-examples/tree/main/server-extension) and then a MIME renderer to stick certain paths into an iframe. 

We should also probably make this part a pip-installable package that's separate, so that it's release cycle is independent of QIIME 2.

What's still a little unclear to me is how an IPython repr invokes a MIME type handler. But surely that's documented _somewhere_.

**Comment by ElDeveloper** [comment]
> We should also probably make this part a pip-installable package that's separate, so that it's release cycle is independent of QIIME 2.

One advantage of this approach is that the server extension would work independently of the kernel in use. In my setup, I run a Jupyter server from the base Conda environment `(base) $`, and I have separate kernels installed for various Conda environments (for example, one for metagenomics, one for amplicon analysis, one for database development, one for network modeling, and so on). Currently, I can't use the Jupyter server extension within QIIME2 environments because the base environment lacks access to the QIIME2 package. This limitation prevents the necessary handlers from being added to the Tornado server.

**If this package became independent of QIIME2, I could enable it by simply installing the new package in the base environment.**

The only remaining task would be to adjust the QZV loader so it no longer depends on the QIIME2 SDK. Instead, it could be designed to directly unzip the archive, bypassing the need for QIIME2-specific functionality.

**Comment by gregcaporaso** [comment]
@ebolyen and I just discussed, and this idea dovetails nicely with our plans in this release cycle to make QIIME 2 View more portable (https://github.com/qiime2/q2view/issues/23). 

@ElDeveloper, we love this idea for better Jupyter Lab integration, but we don't have the ability to prioritize that right now. If we worked on the viewing functionality on the QIIME 2/QIIME 2 View end, would you be interested/available to work on the Jupyter Lab integration aspects?

**Comment by ElDeveloper** [comment]
I agree! https://github.com/qiime2/q2view/issues/23 will fit very nicely!

Let me see what I can do, I'll keep this thread posted! 👍
